pkg/kubelet/lifecycle/handlers.go

```diff
@@ -123,8 +124,17 @@ func (hr *HandlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 			return "", err
 		}
 	}
-	url := fmt.Sprintf("http://%s/%s", net.JoinHostPort(host, strconv.Itoa(port)), handler.HTTPGet.Path)
-	resp, err := hr.httpGetter.Get(url)
+	path := handler.HTTPGet.Path
+	u, err := url.Parse(path)
```

Comment:
Don't start by parsing, that changes the meaning of the field from just a path to allowing it to specify a full URL. Construct the URL object as you do below in all cases